pkg/collector/process.go

```diff
@@ -133,7 +139,8 @@ func (cp *collectingProcess) decodePacket(packetBuffer *bytes.Buffer) (*entities
 		}
 		message.Set = set
 	}
-	cp.messages = append(cp.messages, &message)
+	// the thread(s)/client(s) executing the code will get blocked until the message is consumed/read in other goroutines.
```

Comment:
I assume the boolean useMsgChan will be checked here?

Reply:
We removed the `useMsgChan` because we will create a message channel to output our messages anyways. `useMsgChan` was used for cases that users want to get message list instead of from channel. But we decide to remove it since users can get message list directly from the message channel.

Reply:
Got it. Another question, who will be the consumer of this messageChan, intermediate process?
And I assume there should be only one consumer, is that right? Thanks.

Reply:
Yes, there can be only one consumer. It can be either a standalone collector or the intermediate process but not both.